xarray/core/duck_array_ops.py

```diff
@@ -133,15 +122,20 @@ def fail_on_dask_array_input(values, msg=None, func_name=None):
     "masked_invalid", eager_module=np.ma, dask_module="dask.array.ma"
 )
 
-# sliding_window_view will not dispatch arbitrary kwargs (automatic_rechunk),
-# so we need to hand-code this.
-sliding_window_view = _dask_or_eager_func(
-    "sliding_window_view",
-    eager_module=np.lib.stride_tricks,
-    dask_module=dask_array_compat,
-    dask_only_kwargs=("automatic_rechunk",),
-    numpy_only_kwargs=("subok", "writeable"),
-)
+
+def sliding_window_view(array, window_shape, axis=None, **kwargs):
+    # TODO: some libraries (e.g. jax) don't have this, implement an alternative?
```

Comment:
This is one of the biggest outstanding bummers of wrapping jax arrays. There is apparently [openness](https://github.com/jax-ml/jax/issues/11354) to adding this as an API (even though it would not offer any performance benefit in XLA). But given this is way outside the API standard, whether it makes sense to implement a general version within xarray that doesn't rely on stride tricks.

Reply:
I have something that works pretty well with [this](https://github.com/jax-ml/jax/issues/3171#issuecomment-631826501) style of gather operation. But only in a `jit` context where XLA can work its magic. So I guess this is better left to the specific library to implement, or the user.